File: src/test/resources/pl/msulima/vistula/transpiler/loop.js
const A = XS.rxFlatMap($arg => (vistula.zip($arg.map(X => {
    return X.rxMap($arg => ($arg + 2));
}))));
const B = vistula.constantObservable([
    vistula.constantObservable(1),
    vistula.constantObservable(2)
]).rxFlatMap($arg => (vistula.zip($arg.map(X => {
    const Y = X.rxMap($arg => ($arg + 2));
    return Y;
}))));

File: src/main/scala/pl/msulima/vistula/transpiler/Loop.scala
package pl.msulima.vistula.transpiler

import pl.msulima.vistula.parser.Ast
import pl.msulima.vistula.util.Indent

object Loop {

  def apply: PartialFunction[Ast.stmt, String] = {
    case Ast.stmt.For(Ast.expr.Name(Ast.identifier(name), Ast.expr_context.Load), iterExpr, body, _) =>
      val iter = Transpiler(Ast.stmt.Expr(iterExpr))

      val map =
        s"""vistula.zip($$arg.map($name => {
            |${Indent.leftPad(Transpiler.returnLast(body))}
            |}))""".stripMargin

      s"""$iter.rxFlatMap($$arg => ($map))""".stripMargin
  }
}
